src/client/datascience/notebook/notebookEditorProvider.ts

```diff
@@ -131,28 +132,15 @@ export class NotebookEditorProvider implements INotebookEditorProvider {
     }
     @captureTelemetry(Telemetry.CreateNewNotebook, undefined, false)
     public async createNew(options?: { contents?: string; defaultCellLanguage: string }): Promise<INotebookEditor> {
-        const nbJson: nbformat.INotebookContent = {
-            cells: [],
-            metadata: { orig_nbformat: defaultNotebookFormat.major },
-            nbformat: defaultNotebookFormat.major,
-            nbformat_minor: defaultNotebookFormat.minor
-        };
-        if (options?.contents) {
-            Object.assign(nbJson, JSON.parse(options.contents));
-        }
-        if (options?.defaultCellLanguage) {
-            if (!nbJson.metadata) {
-                nbJson.metadata = nbJson.metadata || { orig_nbformat: 4 };
-            }
-            if (nbJson.metadata.language_info) {
-                nbJson.metadata.language_info.name = options.defaultCellLanguage;
-            } else {
-                nbJson.metadata.language_info = { name: options.defaultCellLanguage };
+        // contents will be ignored
+        const language = options?.defaultCellLanguage ?? PYTHON_LANGUAGE;
+        const cell = new NotebookCellData(NotebookCellKind.Code, '', language);
+        const data = new NotebookData([cell]);
+        data.metadata = {
+            language_info: {
+                name: language
```

Comment:
The metadadata is invalid, we need to set the other values as well (`cells`, `nbformat`, etc).
If we're setting the metadata it has to be a whole, not a partial (in other parts we access the metadata and its type is `nbformat.INotebookContent` and none of them are optional, hence `cells` must exist.

Also, the metadata is not stored in the right place. using `data.metadata` is not correct.
I think we used to set it in custom as follows:
`data.metadata = {custom: nbJson}`

**Background:** In the past the only way to store metadta was to use the `metadata.custom` property, only later was the VS Code API changed to allow any arbitrary properties. but we didn't change that as others were already using this & we didn't want to break it (existing code & .NET extension)

I think this will fix it
We should instead have 
```typescript
const nbJson: nbformat.INotebookContent = {
    cells: [],
    metadata: { orig_nbformat: defaultNotebookFormat.major },
    nbformat: defaultNotebookFormat.major,
    nbformat_minor: defaultNotebookFormat.minor
};
data.metadata = {custom: nbJson}
```

Reply:
We should probably revisit the custom metadata property. I assume that this was from when .NET Interactive built on top of the Jupyter extension for creating new Jupyter files. I think .NET Interactive would ideally depend on the serializer extension, because the custom property sounds like a workaround that is no longer necessary. The New File... contribution point shipped in the 1.59 release so I believe it would also be possible for .NET Interactive to contribute to that directly https://code.visualstudio.com/updates/v1_59#_contribution-point-for-new-file-menu